src/utils/tab.js

```diff
@@ -150,10 +150,12 @@ tabUtils.broadCastTo = (id, msg, isSiteInsideFrame) => {
  * @param  {String}  msg
  * @param  {Boolean} isSiteInsideFrame
  */
-tabUtils.sendMessage = (target, msg, isSiteInsideFrame) => {
+tabUtils.sendMessage = (target, msg, isSiteInsideFrame, name) => {
   let origin = tabUtils.config.origin || '*';
-
-  if (isSiteInsideFrame && target.ref[0]) {
+  if (isSiteInsideFrame && name) {
+    const refIndex = target.ref.length > 1 ? 1 : 0;
```

Comment:
We should add comments in the file for the reasons i.e. Why adding name changes the behavior of target ref